test/broadcasting.jl

```diff
@@ -0,0 +1,221 @@
+module TestBroadcasting
+
+using Test, DataFrames
+
+refdf = DataFrame(ones(3, 5))
+
+@testset "normal data frame and data frame row in broadcasted assignment - one column" begin
+    df = copy(refdf)
+    df[1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test all(Matrix(df[2:end]) .== ones(size(df[2:end])...))
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test all(Matrix(dfv[2:end]) .== ones(size(dfv[2:end])...))
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[end-1:end] .= 10
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test all(Matrix(df[2:end]) .== ones(size(df[2:end])...))
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test all(Matrix(dfv[2:end]) .== ones(size(dfv[2:end])...))
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[end-1:end] .= [10, 10]
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    dfv = @view df[1:2, 2:end]
+    dfr = df[1, 3:end]
+    @test_throws DimensionMismatch df[1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[1] .= rand(2, 1)
+    @test_throws DimensionMismatch dfr[end-1:end] .= rand(3, 1)
+
+    df = copy(refdf)
+    df[:x1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test all(Matrix(df[2:end]) .== ones(size(df[2:end])...))
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:x2] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test all(Matrix(dfv[2:end]) .== ones(size(dfv[2:end])...))
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[[:x4, :x5]] .= 10
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:x1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test all(Matrix(df[2:end]) .== ones(size(df[2:end])...))
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:x2] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test all(Matrix(dfv[2:end]) .== ones(size(dfv[2:end])...))
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[[:x4, :x5]] .= [10, 10]
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    dfv = @view df[1:2, 2:end]
+    dfr = df[1, 3:end]
+    @test_throws DimensionMismatch df[:x1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[:x2] .= rand(2, 1)
+    @test_throws DimensionMismatch dfr[[:x4, :x5]] .= rand(3, 1)
+end
+
+@testset "normal data frame and data frame row in broadcasted assignment - two columns" begin
+    df = copy(refdf)
+    df[[1,2]] .= Matrix(df[[1,2]]) .+ 1
```

Comment:
`df[[1,2]] .= df[[1,2]] .+ 1` does not work because we have just decided to move a decision what to do with broadcasting of a data frame to the other PR. Currently data frame is not supported in broadcasting (this PR makes it only to support broadcasted assignment).

Please comment in https://github.com/JuliaData/DataFrames.jl/issues/1806 what you think should be a proper design when you have time.